content/docs/04.5-devx/03-cluster-groups/35-ingress-cluster-group.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Set Up Ingress"
+metaTitle: "Set Up Ingress for Cluster Groups"
+metaDescription: "Learn how to set up a Kubernetes Ingress for a Palette Cluster Group"
+hideToC: false
+fullWidth: false
+---
+
+import Tabs from 'shared/components/ui/Tabs';
+import InfoBox from 'shared/components/InfoBox';
+import WarningBox from 'shared/components/WarningBox';
+import PointsOfInterest from 'shared/components/common/PointOfInterest';
+import Tooltip from "shared/components/ui/Tooltip";
+
+# Overview
+
+Cluster Groups may have a cluster endpoint type of either Load Balancer or Ingress. The cluster endpoint type determines how Palette Virtual Clusters deployed in a Cluster Group are exposed. You specify the cluster endpoint in Cluster Group Settings.
+
+Using **Ingress** as the cluster endpoint type is a more cost effective way to access your Kubernetes workloads than using type **Load Balancer**, which requires a new cloud load balancer to be provisioned for each virtual cluster.
+
+When you enable **Ingress** as the endpoint for a Cluster Group, you must deploy an NGINX Ingress Controller add-on profile on each host cluster in the Cluster Group. The NGINX Ingress Controller provides the necessary routing functionality for external traffic to reach the Kubernetes API server of each virtual cluster, as well as any apps each virtual cluster contains.
```

Comment:
@karl-cardenas-coding you are right - any ingress controller should theoretically work. The NGINX Ingress Controller just happens to be the most popular option and the only one that we've officially certified.

We should definitely mention that any ingress controller works - provided that it offers support for the following configuration requirements:
1. SSL passthrough enabled (i.e., the ingress controller must be configured to not terminate TLS)
<img width="1473" alt="image" src="https://user-images.githubusercontent.com/1795270/213610331-dd8e3114-9159-4fed-849a-4c96e573f4ab.png">
2. A [TCP service configuration](https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services) to re-route inbound requests on port 6443 to port 443
<img width="1473" alt="image" src="https://user-images.githubusercontent.com/1795270/213610365-d0bbd58d-d0e1-4f21-a3d7-97056443d77f.png">

Reply:
- Requirement 1 is absolutely mandatory. Without SSL passthrough enabled, the TLS handshake will fail when users attempt to connect to their Virtual Cluster(s).
- Requirement 2 is only necessary if the user wishes to terminate TLS for their Apps at the cloud load balancer, but while still retaining the ability to connect to the API server of their Virtual Cluster(s). If users are comfortable using Cert Manager to issue certificates for their Apps' Ingresses within the cluster, they can get away without the TCP service configuration. It just means extra configuration overhead to do all the certificate management in-cluster vs. offloading 99% of it to the cloud lb.

Reply:
@TylerGillson @karl-cardenas-coding 

**Reworded line 21 as follows to clarify any Ingress Controller:** 
"...you must deploy an Ingress Controller add-on profile, such as NGINX, on each host cluster in the Cluster Group. The Ingress Controller provides the necessary routing functionality..." 

**Added these two prerequisites with some explanation:**
- The Ingress Controller must have Secure Socket Layer (SSL) passthrough enabled so that Transport Layer Security (TLS) is not terminated.

    ```yml
    extraArgs:        
  enable-ssl-passthrough: true
    ```

    Palette provides the ```ingress-nginx-host-cluster``` add-on profile with SSL passthrough already enabled.

- If you are not using Palette's ```ingress-nginx-host-cluster``` add-on profile, which reroutes inbound requests from port 6443 to port 443 to use the cloud Load Balancer, you can add the following to the profile you're using: 

    ```yml
    tcp:   
      6443: "nginx/nginx-ingress-controller:443"  
    # 8080: "default/example-tcp-svc:9000"
    ```

    This will terminate TLS for your Apps at the cloud Load Balancer while still allowing you to connect to the API servers of your Virtual Clusters. Otherwise, you must use Cert Manager to issue certificates for App Ingress within the cluster.
```

Reply:
@ritawatson the configuration examples you've provided would only work the NGINX Ingress Controller Helm chart. If someone else was trying to follow along and use, say, the Kong ingress controller, it wouldn't work.

Slight re-wordings:
> The ingress controller must have Secure Socket Layer (SSL) passthrough enabled so that Transport Layer Security (TLS) is not terminated at the ingress controller.

> Palette's `ingress-nginx-host-cluster` add-on profile automatically reroutes inbound requests on port 6443 to port 443 using a [TCP service configuration](https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services). That way, TLS termination (on port 443) for all Apps can occur at the cloud load balancer, while simultaneously allowing connections to the API servers of your Virtual Clusters on port 6443. If you are using an ingress controller other than the NGINX Ingress Controller and would like to terminate TLS at your ingress controller's cloud load balancer, an equivalent TCP service configuration would be required. Alternatively, you may handle all TLS termination in-cluster by configuring Cert Manager to issue a certificate for each App's Ingress.

Reply:
@ritawatson I'd be happy to discuss this on a call tomorrow, but as of right now I wouldn't consider this resolved.

Reply:
Added these revisions and kept the port reroute example with added clarification it's for nginx only and users would include their own ingress controller.